src/coreclr/jit/morph.cpp

```diff
@@ -13160,6 +13148,57 @@ GenTree* Compiler::fgMorphMultiOp(GenTreeMultiOp* multiOp)
 }
 #endif // defined(FEATURE_SIMD) || defined(FEATURE_HW_INTRINSICS)
 
+//------------------------------------------------------------------------
+// fgMorphModToZero: Transform 'a % 1' or 'a % -1' into the equivalent '0'.
+//
+// Arguments:
+//    tree - The GT_MOD/GT_UMOD tree to morph
+//
+// Returns:
+//    The morphed tree, will be a GT_COMMA or a zero constant node.
+//
+GenTree* Compiler::fgMorphModToZero(GenTreeOp* tree)
+{
+    JITDUMP("\nMorphing MOD/UMOD [%06u] to Zero\n", dspTreeID(tree));
+
+    assert(tree->OperIs(GT_MOD, GT_UMOD));
+    assert(tree->gtOp2->IsIntegralConst(1) || tree->gtOp2->IsIntegralConst(-1));
+
+    GenTree* op1 = tree->gtGetOp1();
+    GenTree* op2 = tree->gtGetOp2();
+
+    const var_types type = tree->TypeGet();
+
+    GenTree* zero = gtNewZeroConNode(type);
+
+    GenTree* op1SideEffects = nullptr;
+    gtExtractSideEffList(op1, &op1SideEffects, GTF_ALL_EFFECT);
+    if (op1SideEffects != nullptr)
+    {
+        GenTree* comma = gtNewOperNode(GT_COMMA, type, op1SideEffects, zero);
+
+#ifdef DEBUG
+        // op1 may already have been morphed, so unset this bit.
+        op1SideEffects->gtDebugFlags &= ~GTF_DEBUG_NODE_MORPHED;
```

Comment:
It would be better if we can factor the code so that we don't morph the same trees multiple times (I'm not sure it is in general ok to morph the same node twice). In what cases have we already morphed `op1SideEffects`? Seems odd since this runs in pre-order.

Reply:
Well, that's unfortunate. I wonder if we could chnage `gtBuildCommaList` to only set the flag if the expression being added has already been morphed.

Reply:
Yea there seems to be a case where op1 is a GT_COMMA that is already marked as morphed. I'm not familiar with how that happens, so this is acting like a bandaid.

Reply:
> But what if there are more than two side effects?

Was this addressed @TIHan? I'm seeing failures in Fuzzlyn on main that look related to this change, e.g.
```csharp
// Generated by Fuzzlyn v1.5 on 2022-12-10 12:51:10
// Run on Arm64 MacOS
// Seed: 5902439178145702891
// Reduced from 176.0 KiB to 0.3 KiB in 00:01:07
// Hits JIT assert in Release:
// Assertion failed '((tree->gtDebugFlags & GTF_DEBUG_NODE_MORPHED) == 0) && "ERROR: Already morphed this node!"' in 'Program:Main(Fuzzlyn.ExecutionServer.IRuntime)' during 'Morph - Global' (IL size 34; hash 0xade6b36b; FullOpts)
// 
//     File: /Users/runner/work/1/s/src/coreclr/jit/morph.cpp Line: 12916
// 
public class Program
{
    public static int[][][] s_10;
    public static ulong[][] s_14;
    public static byte[] s_26;
    public static void Main()
    {
        ulong vr6 = s_14[0][0];
        var vr5 = ((short)vr6 * (s_10[0][0][0] | s_26[0])) % 1;
    }
}
```

Maybe try the fix I suggested above?